packages/wrangler/src/bundle.ts

```diff
@@ -16,6 +17,33 @@ type BundleResult = {
   stop: (() => void) | undefined;
 };
 
+/**
+ * Searches for any uses of node's builtin modules, and throws an error if it
+ * finds anything. This plugin is only used when nodeCompat is not enabled.
+ * Supports both regular node builtins, and the new "node:<MODULE>" format.
+ */
+const checkForNodeBuiltinsPlugin = {
+  name: "checkForNodeBuiltins",
+  setup(build: esbuild.PluginBuild) {
+    build.onResolve(
+      {
+        filter: new RegExp(
+          "^(" +
+            builtinModules.join("|") +
+            "|" +
+            builtinModules.map((module) => "node:" + module).join("|") +
+            ")$"
+        ),
+      },
+      () => {
+        throw new Error(
```

Comment:
Should use the logger Error utility

Reply:
We went it to throw an error though, not just log

Reply:
`throw` the `logger.error` or we should add the ability to pass a option that has the logger `throw Error` seems like it should somehow be consistent and utilize the utility, I will unblock the PR either way.

Reply:
I don't understand, the logger only logs, but we want this to be an exception. We log the error to the terminal the error when it gets caught anyway. Where are we missing consistency?

Reply:
> I don't understand, the logger only logs, but we want this to be an exception. We log the error to the terminal the error when it gets caught anyway. Where are we missing consistency?

`logger.error` seems redundant then, you can also throw anything. Consistency is lost in how we console out log errors vs exceptions errors, one has a custom format the other is default system format. Not causing a SIGTERM for some console errors makes sense I suppose maybe certain validation so you can collect all the errors before SIGTERM, I just don't understand why they need to look different too?

Reply:
Wait maybe I missed this, 
> We log the error to the terminal the error when it gets caught anyway.

All Errors are now passed into the `logger.error`?

Reply:
We do! https://github.com/cloudflare/wrangler2/blob/a6e20571e3f7ca2b5a21a0140c9a938120447b2d/packages/wrangler/src/index.tsx#L2653-L2671

Reply:
@rozenmd could you add a small screenshot of what the error looks like?

Reply:
@threepointone: 
<img width="1035" alt="image" src="https://user-images.githubusercontent.com/3822106/168762358-add10043-c36d-4ca8-8a57-34e55be195e1.png">